pkg/utils/utils.go

```diff
@@ -49,3 +52,76 @@ func IsZero[C comparable](val C) bool {
 	var zero C
 	return zero == val
 }
+
+// JustError takes a tuple and returns the last entry, the error.
+func JustError(_ interface{}, err error) error {
+	return err
+}
+
+// WrapIfError decorates an error with the given message.  It is intended to
+// be used with `defer` statements, like so:
+//
+//	func SomeFunction() (err error) {
+//	    defer WrapIfError(&err, "error in SomeFunction:")
+//
+//	    ...
+//	}
+func WrapIfError(err *error, msg string) {
+	if *err != nil {
+		*err = errors.Wrap(*err, msg)
```

Comment:
Oh maybe you don't need package errors:
```suggestion
		*err = fmt.Errorf("%s: %w", msg, *err)
```